1xx-enable-ssl-for-kafka-bridge.md

```diff
@@ -0,0 +1,148 @@
+        certificate: public.crt
+        key: private.key
+...
+```
+The liveness and readiness probes will connect to port 8081 (instead `http.port` or 8080). If users have metrics set up, they would need to use port 8081 to connect to the `metrics` endpoint and cannot reconfigure this port number as it's only for internal use.
+
+An example of KafkaBridge CR with other possible SSL configurations:
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: KafkaBridge
+metadata:
+  name: my-bridge
+spec:
+  replicas: 3
+  bootstrapServers: <cluster_name>-cluster-kafka-bootstrap:9092
+  http:
+    port: 8443
+    tls: #NEW FIELD
+      serverCertChainAndKey: #NEW FIELD
+        secretName: my-secret
+        certificate: public.crt
+        key: private.key
+      config: #NEW FIELD
+        ssl.enabled.protocols: "TLSv1.1, "TLSv1.2", "TLSv1.3"
+        ssl.enabled.cipher.suites: "TLS_DHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384"
+...
+```
+
+## Future Improvements
+
+### Support more formats for the keystore
+We can support formats such as `PKSC12` and `JKS` as they are common for Java applications. If we support these formats, then we would need to add the following configurations:
+- http.ssl.keystore.type
+- http.ssl.keystore.password
+
+### Support mTLS
+In the future, we can support mutual TLS for the client connections to HTTP Bridge as well. This could be implemented by adding a field `trustedCertificates`, for example:
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: KafkaBridge
+metadata:
+  name: my-bridge
+spec:
+  replicas: 3
+  bootstrapServers: <cluster_name>-cluster-kafka-bootstrap:9092
+  http:
+    port: 8443
+    tls: 
+      serverCertChainAndKey: 
+        secretName: my-secret
+        certificate: public.crt
+        key: private.key
+      trustedCertificates: #NEW FIELD
+        - secretName: my-cluster-cluster-cert
+          pattern: "*.crt"
+        - secretName: my-cluster-cluster-cert
+          pattern: "*.crt"    
+...
+```
+This is not in the scope of this proposal, as we will likely to support multiple authentication mechanisms in the future.
+
+In the future, we are also likely to implement authentication and authorization for HTTP Bridge. By having separate listeners, the internal endpoints will not be affected by such changes and will make it easier to implement them.
+
+### Automating certificate management
+
+Users can choose to allow the operator to manage the certificate instead of providing their own certificate and key. This can be implemented by adding another boolean field on the KafkaBridge CR such as `generateCertificate`. The operator then could issue the certificate using the same CA for their Kafka cluster as we do today for internal components such as Cruise Control and Kafka Exporter, however with the upcoming support for cert-manager, the way the operator manages certificates will change. This is the reason we are leaving this feature for future improvement.
```

Comment:
```suggestion
Users can choose to allow the operator to manage the certificate instead of providing their own certificate and key. This can be implemented by adding another boolean field on the `KafkaBridge` CR such as `generateCertificate`. The operator could issue certificates using the same CA for their Kafka cluster, as is currently done for internal components like Cruise Control and Kafka Exporter. However, with upcoming support for cert-manager, certificate management is expected to change. For this reason, we are deferring this feature for implementation in the future.
```